homeassistant/components/romy/config_flow.py

```diff
@@ -0,0 +1,151 @@
+"""Config flow for ROMY integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import romy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_UUID
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN, LOGGER
+
+
+def _schema_with_host() -> vol.Schema:
+    return vol.Schema(
+        {
+            vol.Required(CONF_HOST): cv.string,
+        },
+    )
+
+
+def _schema_with_password() -> vol.Schema:
+    return vol.Schema(
+        {vol.Required(CONF_PASSWORD): vol.All(cv.string, vol.Length(8))},
+    )
+
+
+class RomyConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle config flow for ROMY."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Handle a config flow for ROMY."""
+        self.romy_info: dict[str, Any] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle the user step."""
+        errors: dict[str, str] = {}
+        data = _schema_with_host()
+
+        if user_input:
+            self.romy_info[CONF_HOST] = user_input[CONF_HOST]
+
+            new_romy = await romy.create_romy(self.romy_info[CONF_HOST], "")
+
+            if not new_romy.is_initialized:
+                errors[CONF_HOST] = "cannot_connect"
+            else:
+                # check if already setuped
+                await self.async_set_unique_id(new_romy.unique_id)
+                self._abort_if_unique_id_configured()
+
+                self.romy_info[CONF_NAME] = new_romy.name
+
+                if not new_romy.is_unlocked:
+                    return await self.async_step_unlock_http_interface()
+
+            if not errors:
+                return await self._async_step_finish_config()
+
+        return self.async_show_form(step_id="user", data_schema=data, errors=errors)
+
+    async def async_step_unlock_http_interface(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Unlock the http interface with password if robot is locked."""
+        errors: dict[str, str] = {}
+        data = _schema_with_password()
+
+        if user_input:
+            self.romy_info[CONF_PASSWORD] = user_input[CONF_PASSWORD]
+            new_romy = await romy.create_romy(
+                self.romy_info[CONF_HOST], self.romy_info[CONF_PASSWORD]
+            )
+
+            if not new_romy.is_initialized or not new_romy.is_unlocked:
+                errors[CONF_PASSWORD] = "invalid_auth"
```

Comment:
I think you probably need an explicit `cannot_connect` error again in the case this fails to be initialized

Reply:
saw that too but didn't care/handle it cause its a real small corner case imho, roboter must exaclty went offline in this moment

go back to step user in case of romy uninitialized error